### ISSUE TITLE
Instruction "pylint: disable=deprecated-typing-alias" does not work in specific case

### ISSUE THREAD

**Issue by lorien** [PyCQA/pylint#7928]
### Bug description

This is the exact code to reproduce the bug. The try/except block is required to see bug. Bug happens when I use "disable:" comment in "else" branch.
```python
import typing
from typing import cast

def test():
    try:
        pass
    except IndexError:
        return lambda: None
    else:
        # pylint: disable=deprecated-typing-alias
        return cast(typing.Callable, lambda: None)
```

### Command used

```shell
pylint --load-plugins=pylint.extensions.typing test.py
```

### Pylint output

```shell
$ pylint --load-plugins=pylint.extensions.typing test.py
************* Module test
test.py:12:20: W6001: 'typing.Callable' is deprecated, use 'collections.abc.Callable' instead (deprecated-typing-alias)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 8.75/10, +0.00)
```

### Expected behavior

I expect pylint do not report error "deprecated-typing-alias" because I have explicitly disable it.

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.10.8 (main, Nov  4 2022, 09:21:25) [GCC 12.2.0]
```

**Comment by DanielNoord** [comment]
Duplicate of https://github.com/PyCQA/pylint/issues/872. You can follow that issue to keep updated 😄
